### ISSUE TITLE
A few questions about the AAL module

### ISSUE THREAD

**Issue by jeahun00** [HVision-NKU/ImageCritic#6]
Thank you for your great work and for releasing the code.

I have a few questions I would like to ask.

According to the paper (Section 4.2):

> "We apply the Attention Alignment Loss **only at the double stream blocks** of the model, and $n_l$ is set to the number of double-stream block layers only."

However, in the current implementation, the attention alignment loss ($L_G$, $L_R$) is collected from **all** Attention modules, including both double stream blocks (19 layers) and single stream blocks (38 layers) — totaling 57 layers.

### Root Cause

**1. Both processors store attention maps:**

- `MultiDoubleStreamBlockLoraProcessorWithLoss` (double stream) stores `attn.attention_probs_query_a_key_noise` (layers.py)
- `MultiSingleStreamBlockLoraProcessorWithLoss` (single stream) also stores `attn.attention_probs_query_a_key_noise` (layers.py)

**2. The training loop does not filter by block type:**

```python
# train_ours.py L2642
for name, module in transformer.named_modules():
    if isinstance(module, Attention):  # ← no distinction between double and single
        attn_maps_query_a_key_noise = getattr(module, "attention_probs_query_a_key_noise", None)
        ...
```

`isinstance(module, Attention)` iterates over **all** Attention modules without filtering, so the loss is accumulated from all 57 layers instead of just the 19 double stream layers. If it is correct that the loss is applied only to certain streams as mentioned in the paper, I would like to cautiously suggest that the code should be modified as in the following suggestion.

### Suggested Fix

Filter by module name in the training loop:

```python
for name, module in transformer.named_modules():
    if isinstance(module, Attention) and "transformer_blocks" in name and "single" not in name:
        ...
```

This restricts the attention alignment loss to `transformer_blocks.{0-18}` (double stream) only, matching the paper.

**Comment by ouyangziheng** [comment]
https://github.com/HVision-NKU/ImageCritic/blob/a2b1a04399663114caddcf84c12019f3226a64ef/src/layers.py#L305

**Comment by ouyangziheng** [comment]
Thank you for your attention to this detail. In our current single-stream implementation of MultiSingleStreamBlockLoraProcessorWithLoss(nn.Module), the code path for extracting and storing the attention maps has been commented out—specifically, the logic that slices query_cond_a, query_cond_b, and key_noise, computes the corresponding attention probabilities, and assigns them back to attn.
